### PR TITLE
Avoid copying values when possible

### PR DESCRIPTION
**Pull request by jbevain**

### CODE REVIEW

File: src/Parlot/Fluent/ParseResult.cs [hunk 1]
```diff
@@ -31,6 +31,6 @@ public void Set(string buffer, in TextPosition start, in TextPosition end, strin
     
     public static class ParseResultExtensions
     {
-        public static ReadOnlySpan<char> GetSpan<T>(this ParseResult<T> result) => result.Buffer.AsSpan(result.Start.Offset, result.End - result.Start);
+        public static ReadOnlySpan<char> GetSpan<T>(this in ParseResult<T> result) => result.Buffer.AsSpan(result.Start.Offset, result.End - result.Start);
```
Comment (lahma) [inline src/Parlot/Fluent/ParseResult.cs:34]: IIRC `in` shouldn't be used with regular structs, only with read-only ones. `ParseResult` is regular.

Comment (jbevain) [reply src/Parlot/Fluent/ParseResult.cs:34]: @lahma good catch! I'll remove this one!

Comment (sebastienros) [reply src/Parlot/Fluent/ParseResult.cs:34]: This whole method/class can be removed. And the `Buffer` in `ParseResult` too.

Comment (sebastienros) [reply src/Parlot/Fluent/ParseResult.cs:34]: But it's out of the scope of this PR

Comment (jbevain) [reply src/Parlot/Fluent/ParseResult.cs]: Yeah I saw it wasn't used in Parlot but didn't check if you were using it outside of it.

**Pull request merged**